pkg/vm/stackitem/item.go

```diff
@@ -826,12 +826,14 @@ func NewMap() *Map {
 
 // NewMapWithValue returns a new Map object filled with the specified value.
 func NewMapWithValue(value []MapElement) *Map {
-	if value != nil {
-		return &Map{
-			value: value,
+	res := NewMap()
+	for i := range value {
+		if err := IsValidMapKey(value[i].Key); err != nil {
+			panic(err)
```

Comment:
It's too risky to panic inside `NewMapWithValue`, it may be used outside the VM. Either adjust a documentation of this function and don't check for `value` correctness or change the signature of this function to return error. BTW, with the updated PACKMAP, do we really use this function anywhere?

Reply:
No, `NewMapWithValue` is used only in tests.